frontend/src/components/layouts/AppLayout.tsx

```diff
@@ -18,14 +22,29 @@ export default function AppLayout() {
 
   return (
     <>
-      <div className="font-sans min-h-screen w-full flex flex-col">
-        <SidebarProvider>
-          <AppSidebar />
-          <SidebarInset>
-            <div className="flex flex-1 flex-col gap-4 p-4">
-              <Outlet />
-            </div>
-          </SidebarInset>
+      <div
+        className={cn(
+          "font-sans min-h-screen w-full flex flex-col",
+          showBanner
+            ? "[--header-height:calc(theme(spacing.9))]"
+            : "[--header-height:0]"
+        )}
+      >
+        <SidebarProvider className="flex flex-col">
+          {showBanner && <UpdateBanner onDismiss={dismissBanner} />}
+          <div className="flex flex-1">
+            <AppSidebar />
+            <SidebarInset>
+              <div
+                className={cn(
+                  "flex flex-1 flex-col gap-4 p-4",
+                  showBanner && "mt-14 md:mt-9"
```

Comment:
same here.

Would this stuff be needed if the update banner wasn't fixed position but actually was a normal element rendered on top of everything else?

Reply:
I tried to do that as well, but the sidebar is also fixed so it's quite hard to push it down, that's why I had to calculate the banner height based on how many lines it has (`36px` and `56px`)